simapp/app.go

```diff
@@ -414,6 +432,20 @@ func NewSimApp(
 	return app
 }
 
+// GetInflationWithTime returns the new inflation rate with given time
+func GetInflationWithTime(ctx sdk.Context, m minttypes.Minter, params minttypes.Params, bondedRatio sdk.Dec) sdk.Dec {
+	yearsPassed := (ctx.BlockTime().Sub(genesisTime)).Seconds() / (60 * 60 * 8766)
+
+	// inflation = initialInflation * 2^ -(timePassedfromGenesis/TimetoHalveInflationRate)
+	twoPowerDec, err := sdk.NewDecFromStr(fmt.Sprintf("%.18f", math.Pow(2, -(yearsPassed/tHalf))))
+	if err != nil {
+		panic(err)
+	}
+	m.Inflation = initialInflation.Mul(twoPowerDec)
```

Comment:
Doesn't this also get set to the return value of `GetInflationWithTime()`?

Reply:
Here, we are just updating inflation value of minter struct so that we can pass it to `NextInflationRate()` for bonded ratio consideration.